src/analyzer/optimizations/shift_math.rs

```diff
@@ -40,31 +40,38 @@ fn check_if_inputs_are_power_of_two(
     box_expression_1: Box<Expression>,
 ) -> bool {
     //create a boolean to determine if either of the inputs are a power of two
-    let mut is_even: bool = false;
 
     //if the first expression is a number literal that is a power of 2
     if let Expression::NumberLiteral(_, val_string, _) = *box_expression {
-        let value = val_string
-            .parse::<u32>()
-            .expect("Could not parse NumberLiteral value from string to u32");
-
-        if (value != 0) && ((value & (value - 1)) == 0) {
-            is_even = true;
+        match val_string.parse::<u128>() {
+            Ok(value) => {
+                if (value != 0) && ((value & (value - 1)) == 0) {
+                    return true;
+                }
+            }
+            Err(_) => println!(
+                "Could not parse NumberLiteral value '{}' from string to u128",
+                val_string
+            ),
```

Comment:
Lets use `U256::from_dec_str()` instead of u128, so we can handle number literals larger than u128.